.github/PULL_REQUEST_TEMPLATE/development.md

```diff
@@ -0,0 +1,33 @@
+## Pre-Review Checklist for the PR Author
+
+1. [ ] Branch follows the naming format (`iox-#123-this-is-a-branch`)
+1. [ ] Commits messages are according to this [guideline][commit-guidelines]
+    - [ ] Commit messages have the issue ID (`iox-#123 commit text`)
+    - [ ] Commit messages are signed (`git commit -s`)
+    - [ ] Commit author matches [Eclipse Contributor Agreement][eca] (and ECA is signed)
+2. [ ] Update the PR title
+   - Follow the same conventions as for commit messages
+   - Link to the relevant issue
+3. [ ] Add sensible notes for the reviewer
+4. [ ] Assign PR to reviewer
+
+[commit-guidelines]: https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
+[eca]: http://www.eclipse.org/legal/ECA.php
+
+## Notes for Reviewer
+<!-- Items in addition to the checklist below that the reviewer should look for -->
+
+## Checklist for the PR Reviewer
+
+- [ ] Commits are properly organized and messages are according to the guideline
+- [ ] PR title describes the changes
+- [ ] Approve
+
+## Post-review Checklist for the PR Author
```

Comment:
This probably should become `## Post-review Checklist for the Eclipse Committer` since not all contributors have the right to merge